api/src/test/java/demo/ContextIntegrationTest.java

```diff
@@ -150,4 +173,10 @@ public static void persistCreatedEntities(EntityManager em) {
     createdEntities.forEach(em::persist);
     em.flush();
   }
+
+  protected static void logTestResult (MvcResult result, byte[] bytes) throws UnsupportedEncodingException {
+    System.out.println("Test failed! Logging request and response details:");
```

Comment:
@Shmarkus 
I like the idea of adding additional logging, when a test fails, What would be the best way to implement it?

I did a pre-limenary implementation, but it´s kind of clunky....

Reply:
The failing test shouldn't need additional logging. in theory if the test fails, you should already know what went wrong.